### PR TITLE
Add Facebook and Google authentication

### PR DESCRIPTION
**Pull request by TarasRudyk**

### CODE REVIEW

File: client/components/place/place-list/place-list.component.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ angular.module('placeList', ['filterMapType', 'popularTracks'])
         var typesLength;
 
         this.user = currentUser;
+        console.log(currentUser);
```
Comment (yurykozhenov) [inline client/components/place/place-list/place-list.component.js]: Remove log

File: client/components/place/popular-tracks/popular-tracks.component.js [hunk 2]
```diff
@@ -5,7 +5,6 @@ angular.module('popularTracks', [])
       var self = this;
       self.popularTracks = [];
       var userLocation;
-
```
Comment (yurykozhenov) [inline client/components/place/popular-tracks/popular-tracks.component.js:8]: Return empty line here

File: server/app/models/user.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ var UserSchema = new Schema({
   },
   password: {
     type: String,
-    required: 'Password is required',
+  //  required: 'Password is required',   //has to be optional due facebook etc.,  autho password is not require
```
Comment (yurykozhenov) [inline server/app/models/user.js]: :eyes: Nope, password should always be required.
When user signing up through _Facebook_ or _Google+_ they should also add password by themselves.

File: server/app/models/user.js [hunk 4]
```diff
@@ -52,11 +52,25 @@ var UserSchema = new Schema({
   },
   provider: {
     type: String,
-    required: 'Provider is required'
+    //required: 'Provider is required'  //has to be optional
```
Comment (yurykozhenov) [inline server/app/models/user.js]: Nope, you should add provider by yourself when creating user.

File: server/config/strategies/auth.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
```
Comment (yurykozhenov) [inline server/config/strategies/auth.js]: You should move this file content inside `env/development.js` and `env/production.js`

File: server/config/strategies/facebook-auth.js [hunk 6]
```diff
@@ -1,17 +1,48 @@
...
-    callbackURL: 'http://localhost:' + PORT + '/auth/facebook/callback'
-  },
-  function(accessToken, refreshToken, profile, done) {
-    User.findOrCreate(..., function(err, user) {
-      if (err) { return done(err); }
-      done(null, user);
-    });
+                  user.save(function(err) {
+                    if(err) {
+                      console.log(err);  //handle errors!
```
Comment (yurykozhenov) [inline server/config/strategies/facebook-auth.js]: Remove log.

User should have some feedback if there is an error, `console.log` doesn't give any of it.

File: server/config/strategies/facebook-auth.js [hunk 7]
```diff
@@ -1,17 +1,48 @@
...
+                     } else {
+                      console.log("saving user ...");
```
Comment (yurykozhenov) [inline server/config/strategies/facebook-auth.js]: Remove log

File: server/config/strategies/facebook-auth.js [hunk 8]
```diff
@@ -1,17 +1,48 @@
...
+                      done(null, user);
+                    }
+                  });
+                  console.log(user)
```
Comment (yurykozhenov) [inline server/config/strategies/facebook-auth.js]: Remove log

File: server/config/strategies/google-OAught2.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+                  user.providerData.google.id = profile.id;
+                  user.providerData.google.token = accessToken;
+                  user.email = profile.emails[0].value;
+                  user.avatar = profile.photos[0].value;
+                  user.firstName = profile.name.givenName;
+                  user.lastName = profile.name.familyName;
+
+                  user.save(function(err) {
+                    if(err) {
+                      console.log(err);  //handle errors
```
Comment (yurykozhenov) [inline server/config/strategies/google-OAught2.js]: Remove log.

User should have some feedback if there is an error, console.log doesn't give any of it.

File: server/config/strategies/google-OAught2.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+                     } else {
+                      console.log("saving user ...");
```
Comment (yurykozhenov) [inline server/config/strategies/google-OAught2.js]: Remove log

File: server/config/strategies/google-OAught2.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+                      done(null, user);
+                    }
+                  });
+                  console.log(user)
```
Comment (yurykozhenov) [inline server/config/strategies/google-OAught2.js]: Remove log

File: server/app/models/user.js (hunk 4 above)
Comment (TarasRudyk) [inline server/app/models/user.js]: Ok, added already :-)

**Pull request merged**